### PR TITLE
Fix iobuf inference

### PR DESCRIPTION
**Pull request by litghost**
Attempt to make #1339 pass CI.

### CODE REVIEW

File: xc7/yosys/synth.tcl [hunk 1]
```diff
@@ -2,21 +2,29 @@ yosys -import
...
+    read_verilog -lib -D_EXPLICIT_CARRY +/xilinx/cells_sim.v
+    read_verilog -lib +/xilinx/cells_xtra.v
+
+    # Overwrite some models (e.g. IBUF with more parameters)
+    read_verilog -lib $::env(symbiflow-arch-defs_SOURCE_DIR)/xc7/techmap/iobs.v
+
+    hierarchy -check -auto-top
+
+    # Start flow after library reading
+    synth_xilinx -vpr -flatten -abc9 -nosrl -noclkbuf -nodsp -iopad -run prepare:check
```
Comment (mithro) [inline xc7/yosys/synth.tcl]: Why does this run from `prepare` to `check` while the other pathway runs everything?

File: xc7/yosys/synth.tcl [hunk 2]
```diff
@@ -2,21 +2,29 @@ yosys -import
...
-# Map inferred IBUFs/OBUFs to wires. This way ROI targets will work.
-techmap -map $::env(symbiflow-arch-defs_SOURCE_DIR)/xc7/techmap/iob_make_wires.v
-# Map previously renamed IBUFs/OBUFs back to their former names.
-techmap -map $::env(symbiflow-arch-defs_SOURCE_DIR)/xc7/techmap/iob_unmap.v
+#
+# Do not infer IOBs for targets that use a ROI.
+if { $::env(USE_ROI) == "TRUE" } {
+    synth_xilinx -vpr -flatten -abc9 -nosrl -noclkbuf -nodsp
+} else {
+    # Read Yosys baseline library first.
```
Comment (mithro) [inline xc7/yosys/synth.tcl]: Why are these not needed for the other pathway?

File: xc7/yosys/synth.tcl (hunk 1 above)
Comment (litghost) [reply xc7/yosys/synth.tcl]: Did you read the comment?

File: xc7/yosys/synth.tcl (hunk 2 above)
Comment (litghost) [reply xc7/yosys/synth.tcl]: What do you mean by "these"?

Comment (mithro) [reply xc7/yosys/synth.tcl]: Line 17+18.

File: xc7/yosys/synth.tcl (hunk 1 above)
Comment (mithro) [reply xc7/yosys/synth.tcl]: Oh - I think I understand now.

The first couple of lines in this section do the library reading. Then you start from the prepare stage (which is the first stage after library reading)?

Comment (litghost) [reply xc7/yosys/synth.tcl]: Yes

**Pull request merged**